### PR TITLE
Task-55636 : Add notif download

### PR DESCRIPTION
**Pull request by wbenabdessalem**
Use native system when i download a document.

Given I'm a user in the platform using the mobile app

When i donwload a document

Then the native system should be applied to download it and display the progress of this download.

### CODE REVIEW

File: app/src/main/java/org/exoplatform/tool/JavaScriptInterface.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
+package org.exoplatform.tool;
```
Comment (rdenarie) [inline app/src/main/java/org/exoplatform/tool/JavaScriptInterface.java]: Can you add header

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/exoplatform/tool/JavaScriptInterface.java (hunk 1 above)
Comment (wbenabdessalem) [reply app/src/main/java/org/exoplatform/tool/JavaScriptInterface.java]: Done

File: app/src/main/java/org/exoplatform/tool/JavaScriptInterface.java [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+/*
+ * Copyright (C) 2003-2019 eXo Platform SAS.
```
Comment (rdenarie) [inline app/src/main/java/org/exoplatform/tool/JavaScriptInterface.java]:
```suggestion
 * Copyright (C) 2022 eXo Platform SAS.
```

**Review by rdenarie — CHANGES_REQUESTED** [review]

**Review by rdenarie — CHANGES_REQUESTED** [review]
the copyright is not the good one.
You need to take the AGPL as you are on exoplatform organization

**Comment by wbenabdessalem** [comment]
Can you check again please ?

File: app/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -21,22 +23,38 @@
...
+            android:name="com.google.android.actions"
+            android:resource="@xml/provider_paths" />
         <activity
             android:name=".activity.LauncherActivity"
             android:noHistory="true"
             android:screenOrientation="portrait"
             android:theme="@style/SplashStyle">
             <intent-filter>
                 <action android:name="android.intent.action.MAIN" />
+
```
Comment (rdenarie) [inline app/src/main/AndroidManifest.xml]:
```suggestion
```

File: app/src/main/AndroidManifest.xml [hunk 4]
```diff
@@ -21,22 +23,38 @@
...
                 <category android:name="android.intent.category.LAUNCHER" />
             </intent-filter>
             <intent-filter>
                 <action android:name="android.intent.action.VIEW" />
+
```
Comment (rdenarie) [inline app/src/main/AndroidManifest.xml]:
```suggestion
```

File: app/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -21,22 +23,38 @@
...
                 <category android:name="android.intent.category.DEFAULT" />
                 <category android:name="android.intent.category.BROWSABLE" />
-                <data android:scheme="https"
+
```
Comment (rdenarie) [inline app/src/main/AndroidManifest.xml]:
```suggestion
```

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: app/src/main/AndroidManifest.xml [hunk 6]
```diff
@@ -4,15 +4,17 @@
...
-    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
+    <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
     <uses-permission android:name="android.permission.INTERNET" />
     <uses-permission android:name="android.permission.CAMERA" />
     <uses-permission android:name="android.permission.MODIFY_AUDIO_SETTINGS" />
     <uses-permission android:name="android.permission.MICROPHONE" />
     <uses-permission android:name="android.permission.RECORD_AUDIO" />
-    <uses-feature android:name="android.hardware.microphone"/>
-    <uses-feature android:name="android.hardware.camera"/>
+
```
Comment (rdenarie) [inline app/src/main/AndroidManifest.xml]:
```suggestion
```

File: app/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -4,15 +4,17 @@
...
+    <uses-feature android:name="android.hardware.microphone" />
+    <uses-feature android:name="android.hardware.camera" />
     <uses-feature android:name="android.hardware.camera.any" />
+
```
Comment (rdenarie) [inline app/src/main/AndroidManifest.xml]:
```suggestion
```

**Review by rdenarie — CHANGES_REQUESTED** [review]

**Review by rdenarie — APPROVED** [review]

**Pull request merged**